### PR TITLE
[Gazebo9] Added reflectance callback to publish reflectance image in ROS 

### PR DESCRIPTION
**Pull request by ahcorde**
Moved this PR https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1044 to `noetic-devel` to avoid ABI changes in melodic

Refletance must be an image.

To include the reflectance:

```
<link>
...
  <visual name="visual">
    ...
      <plugin filename="libReflectancePlugin.so" name="reflectance_plugin">
        <reflectance_map>your_reflectance_map.png</reflectance_map>
      </plugin>
      ...
   </visual>
...
</link>
```

### CODE REVIEW

**Comment by iche033** [comment]
could you update the plugin params in the PR description? i.e. it now expects `<reflectance_map>` instead of `<material>`

**Comment by ahcorde** [comment]
@iche033, Do you mean this change https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1046/commits/63bc8ad5bd04981ad811dc88fc405edbd955d88f?

**Comment by iche033** [comment]
I didn't know I can just edit your comment in the PR. I just updated it (the first comment in the conversation)

**Comment by iche033** [comment]
Just checking this is also what you get from `/depth_cam/reflectance` when running `gazebo_ros_depth_camera.world`?

depth_cam_reflectance

The output is a little darker than the actual terrain.png but I wonder if it's how the data is visualized.

**Comment by ahcorde** [comment]
yes, I think it's a visualization matter. I got the same image

**Review by iche033 — APPROVED** [review]

**Comment by ahcorde** [comment]
friendly ping @chapulina

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_depth_camera.h [hunk 1]
```diff
@@ -89,6 +89,10 @@ namespace gazebo
     protected: virtual void OnNewImageFrame(const unsigned char *_image,
                    unsigned int _width, unsigned int _height,
                    unsigned int _depth, const std::string &_format);
+     /// \brief Update the controller
```
Comment (chapulina) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_depth_camera.h]: Please document all the new members and use `override` when possible.

**Review by chapulina — APPROVED** [review]
LGTM, I just have a nitpick about documentation. Also note that the upstream PR must be merged first.

**Review by chapulina — APPROVED** [review]

**Comment by apojomovsky** [comment]
Hi guys, and sorry for breaking in into a long merged PR, but I'm not being able to run this example. Particularly, what I'm doing is:

```
roscore &
rosrun gazebo_ros gazebo ./opt/ros/noetic/share/gazebo_plugins/test/test_worlds/gazebo_ros_depth_camera.world
```
And on a separate terminal I'm running `rqt_image_view`, but there doesn't seem to be a `<prefix>/reflectance` topic being published whatsoever.

Screenshot from 2021-11-10 14-02-54


I'm running ROS noetic with gazebo_ros_pkgs (v 2.9.2-1focal.20211007.215718), all installed from debians via apt.

Any suggestions would be welcome!

**Pull request merged**